src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Private/JavaScriptMarshalImpl.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace System.Runtime.InteropServices.JavaScript.Private
+{
+    internal static partial class JavaScriptMarshalImpl
+    {
+        internal static Function log = new Function("data", @"console.log(data)");
+
+        static JavaScriptMarshalImpl()
+        {
+            toJsMarshallers.Add(typeof(IJSObject).TypeHandle.Value, MarshalJsObjectToJs);
+            toCsMarshallers.Add(typeof(IJSObject).TypeHandle.Value, MarshalJsObjectToCs);
+            toJsMarshallers.Add(typeof(string).TypeHandle.Value, MarshalStringToJs);
+            toCsMarshallers.Add(typeof(string).TypeHandle.Value, MarshalStringToCs);
+
+            //log.Call(null, $"JavaScriptMarshalImpl string {typeof(string).TypeHandle.Value}");
+            //log.Call(null, $"JavaScriptMarshalImpl void {typeof(void).TypeHandle.Value}");
+        }
+
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        internal static extern InvokeJSResult InvokeJSFunction0(string internedFunctionName, IntPtr signature, IntPtr data);
```

Comment:
Since these functions (0, 1, etc) all have the same signature, I don't think we should make them separate icalls.

Reply:
We would have to dispatch dynamically to implementations with different number of convertors on js side. This cost me little on startup and less on runtime.